testing/tests/DevExpress.ui.widgets.dataGrid/virtualScrolling.integration.tests.js

```diff
@@ -6487,20 +6487,24 @@ QUnit.module('Virtual Scrolling', baseModuleConfig, () => {
         this.clock.tick(10);
 
         // act
-        dataGrid.option('editing.editRowKey', 14);
+        dataGrid.option('editing.editRowKey', 12);
```

Comment:
The test had to be edited because after getCellElementsCore returns undefined if RowIndex = -1, the last visible row is no longer focused.

In general, the fact that -1 comes to getCellElementsCore is either an error or a by-design, and this needs to be researched as a separate problem.